src/requestgenerator.cpp

```diff
@@ -337,10 +337,17 @@ QNetworkReply *RequestGenerator::contactMultiget(const QString &serverUrl, const
     QString uriHrefs;
     Q_FOREACH (const QString &uri, contactUris) {
         // note: uriHref is of form: <d:href>/addressbooks/johndoe/contacts/acme-12345.vcf</d:href> etc.
+        QString href = uri.toHtmlEscaped();
+        int lastPathMarker = href.lastIndexOf('/');
+        if (lastPathMarker > 0) {
+            // percent-encode the filename
+            QString vcfName = QUrl::toPercentEncoding(href.mid(lastPathMarker + 1));
+            href = href.mid(0, lastPathMarker) + '/' + vcfName;
```

Comment:
The slash is already there, this could be just "href.left(lastPathMarker + 1) + vcfName"